src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.J.cs

```diff
@@ -0,0 +1,298 @@
+						if (!ParserHelpers.TryGetNextTwoDigits(source, sourceIndex, out sourceIndex, out second))
+                        {
+                            state = Utf8Constants.DateJParserState.Invalid;
+                            break;
+                        }
+
+						if (sourceLength - sourceIndex > 1)
+                        {
+                            if (source[sourceIndex] == Utf8Constants.Period)
+                                state = Utf8Constants.DateJParserState.Fraction;
+                            else if (source[sourceIndex] == Utf8Constants.UtcOffsetChar)
+				            {
+					            offsetChar = Utf8Constants.UtcOffsetChar;
+					            continueParsing = false;
+				            }
+                            else if (source[sourceIndex] == Utf8Constants.Plus)
+                            {
+                                offsetChar = Utf8Constants.Plus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else if (source[sourceIndex] == Utf8Constants.Minus)
+                            {
+                                offsetChar = Utf8Constants.Minus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else
+							{
+								continueParsing = false;
+								break;
+							}
+
+                            sourceIndex++;
+                        }
+						break;
+					case Utf8Constants.DateJParserState.Fraction:
+						while (sourceIndex < sourceLength && ParserHelpers.IsDigit(source[sourceIndex]))
+						{
+                            if (!((fraction * 10) + (int)(source[sourceIndex] - 48u) <= Utf8Constants.MaxDateTimeFraction))
+                            {
+                                sourceIndex++;
+                                break;
+                            }
+
+                            fraction = (fraction * 10) + (int)(source[sourceIndex++] - 48u);
+						}
+
+						if (fraction != 0)
+						{
+							while (fraction * 10 <= Utf8Constants.MaxDateTimeFraction)
+								fraction *= 10;
+			            }
+
+						if (sourceIndex == sourceLength)
+							break;
+						
+						if (source[sourceIndex] == Utf8Constants.UtcOffsetChar)
+						{
+							offsetChar = Utf8Constants.UtcOffsetChar;
+							continueParsing = false;
+							sourceIndex++;
+						}
+						else if (source[sourceIndex] == Utf8Constants.Plus)
+                        {
+                            offsetChar = Utf8Constants.Plus;
+                            state = Utf8Constants.DateJParserState.OffsetHours;
+							sourceIndex++;
+                        }
+                        else if (source[sourceIndex] == Utf8Constants.Minus)
+                        {
+                            offsetChar = Utf8Constants.Minus;
+                            state = Utf8Constants.DateJParserState.OffsetHours;
+							sourceIndex++;
+						}
```

Comment:
I think there are a few whitespace issues in this file, you might want to reformat it.

Reply:
Yes, it's a bit hard to follow with the indentation being non-standard.  Thanks.